data-in-pipeline/app/extract/navigator_document.py

```
@@ -1,38 +1,126 @@
+
+    :param import_id: Document identifier
+    :param config: Typed connector configuration
+    :return ExtractedEnvelope with document data and metadata
 
-def extract_navigator_document(import_id: str) -> Extracted[NavigatorDocument]:
-    print(f"Fetching {import_id} from {API_BASE_URL}/families/{import_id}")
-    data = _fetch(import_id)
-    extacted = Extracted(data=data, source="navigator_document")
-    return extacted
+    Raises:
+        requests.HTTPError: On non-retryable HTTP errors
+        requests.Timeout: If request exceeds timeout
+    """
 
+    logger.info(f"Extracting document {import_id} from Navigator")
 
-def _fetch(id: str) -> NavigatorDocument:
-    """Fetch a document from the API.
+    try:
+        document = _fetch_with_retry(import_id, config)
 
-    :param id: The id of the document.
-    :type id: str
-    :rtype: NavigatorDocument
+        envelope = ExtractedEnvelope(
+            data=document,
+            envelope_id=generate_envelope_uuid(),
+            source_name="navigator-documents",
+            source="navigator",
+            source_record_id=import_id,
+            raw_payload=document.model_dump_json(),
+            content_type="application/json",
+            connector_version="1.0.0",
+            run_id="",  # Retrieve from Prefect context
+            extracted_at=datetime.utcnow(),
+            metadata={
+                "endpoint": f"/families/documents/{import_id}",
+                "http_status": 200,
+            },
+        )
+
+        logger.info(f"Successfully extracted document {import_id}")
+        return envelope
+
+    except requests.HTTPError as e:
+        logger.error(f"HTTP error fetching {import_id}: {e}")
+        raise e
+    except requests.Timeout as e:
+        logger.error(f"Timeout fetching {import_id}: {e}")
+        raise e
+    except Exception as e:
+        logger.error(f"Unexpected error fetching {import_id}: {e}")
+        raise e
```

✨ nice - no swallowing errors!